src/hotspot/os/aix/attachListener_aix.cpp

```diff
@@ -451,6 +451,10 @@ void AixAttachOperation::complete(jint result, bufferedStream* st) {
     // Shutdown the socket in the cleanup function to enable more than
     // one agent attach in a sequence (see comments to listener_cleanup()).
     AixAttachListener::write_fully(this->socket(), (char*) st->base(), st->size());
+    if (st->was_truncated()) {
+      char msg [] = "\n(truncated)\n";
+      AixAttachListener::write_fully(this->socket(), msg, strlen(msg));
+    }
```

Comment:
Isn't this happening too late? The st content has already been truncated and we have written the "final" message to the socket for the client to read, and then we try to send a follow-up message that the original was truncated. ???

Reply:
Okay. Do we not then have to check the rc of the first `write_fully` and only do this follow up write if `rc == 0`?